src/apis.md

```diff
@@ -0,0 +1,165 @@
+# Questions about designing APIs for others
+
+See also the excellent [Rust API guidelines](https://rust-lang.github.io/api-guidelines/about.html).
+The document you're reading aims to provide extra hints which may be especially
+useful to folk coming from C++, but that's the canonical reference.
+
+## When should my type implement `Default`?
+
+Whenever you'd provide a default constructor in C++.
+
+## When should my type implement `From`, `Into` and `TryFrom`?
+
+You should think of these as equivalent to implicit conversions in C++. Just
+as with C++, if there are _multiple_ ways to convert from your thing to another
+thing, don't implement these, but if there's a single obvious conversion, do.
+
+Usually, don't implement `Into` but instead implement `From`.
+
+## How should I expose constructors?
+
+See the previous two answers: where it's simple and obvious, use the standard
+traits to make your object behavior predictable.
+
+If you need to go beyond that, remember you've got a couple of extra toys in Rust:
+
+* A "constructor" could return a `Result<Self>`
+* Your constructors can have names, e.g. `with_capacity`, `pin`
+
+## When should my type implement `AsRef`?
+
+If you have a type which contains another type, provide `AsRef` especially
+so that people can clone the inner type. It's good practice to provide explicit
+versions as well (for example, `String` implements `AsRef<str>` but also
+provides `.as_str()`.)
+
+## When should I implement `Copy`?
+
+> Anything that is integer-like or reference-like should be `Copy`; other things
+> shouldn’t - MY
```

Comment:
Nit: "...shouldn’t be. — MY"

"...uphold. — AH"

I.e. end the quote with punctuation, then em-dash for attribution.

Also, these use curly quotes (yay!) but the rest of the document doesn't. Better to be consistent either way.